src/GitHub/index.jsx

```diff
@@ -0,0 +1,158 @@
+//#region > Imports
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+//> MDB
+// "Material Design for Bootstrap" is a great UI design framework
+import { MDBBtn, MDBIcon } from "mdbreact";
+
+//> PopupWindow
+import PopupWindow from "../PopupWindow";
+//> Utils
+import { toQuery, GuidGenerator } from "../utils";
+//#endregion
+
+//#region > Components
+/**
+ * @class A button that is used for GitHub OAuth
+ */
+class OAuth2Login extends Component {
+  constructor(props) {
+    super(props);
+    this.onBtnClick = this.onBtnClick.bind(this);
+    this.onRequest = this.onRequest.bind(this);
+    this.onSuccess = this.onSuccess.bind(this);
+    this.onFailure = this.onFailure.bind(this);
+  }
+
+  onBtnClick() {
+    const {
+      authorizationUrl,
+      clientId,
+      clientSecret,
+      scope,
+      redirectUri,
+    } = this.props;
+    const search = toQuery({
+      client_id: clientId,
+      scope,
+      redirect_uri: redirectUri,
+      response_type: "code",
+    });
+
+    const popup = PopupWindow.open(`${authorizationUrl}?${search}`);
+    this.popup = popup;
+
+    this.onRequest();
+    popup.then(
+      (data) => this.onSuccess(data),
+      (error) => this.onFailure(error)
+    );
+  }
+
+  onRequest() {
+    const { onRequest } = this.props;
+
+    onRequest();
+  }
+
+  async onSuccess(data) {
+    if (!data.code) {
+      return this.onFailure(new Error("'code' not found"));
+    }
+
+    const AuthorizeUrl =
+      `https://cors.snek.at/` +
+      `https://github.com/login/oauth/access_token?code=${data.code}` +
+      `&client_id=${this.props.clientId}` +
+      `&client_secret=${this.props.clientSecret}` +
+      `&redirect_uri=${this.props.redirectUri}` +
+      `&state=${await GuidGenerator()}`;
```

Comment:
It seems like the scope is missing. Therefore the requests access token has zero privileges.

![image](https://user-images.githubusercontent.com/52858351/85716176-7d5ceb00-b6ec-11ea-95b9-42d35734b605.png)